src/Provider/DoctrineMongoDbProvider.php

```diff
@@ -43,7 +50,7 @@ public function register(Container $container)
             $container['mongodbs.options'] = $tmp;
         });
 
-        $container['mongodbs'] = function ($container) {
+        $container['mongodbs'] = $container->factory(function ($container) {
```

Comment:
you're right, function seems to be enough for this ... I'll check the pimple/container factory placements in this class for simple closure calls also